src/treecodes.py

```diff
@@ -42,11 +43,18 @@
              'mochitest-e10s-browser-chrome-1': {'group': 'M-e10s', 'code': 'bc1'},
              'mochitest-e10s-browser-chrome-2': {'group': 'M-e10s', 'code': 'bc2'},
              'mochitest-e10s-browser-chrome-3': {'group': 'M-e10s', 'code': 'bc3'},
+             'mochitest-browser-chrome-e10s-1': {'group': 'M-e10s', 'code': 'bc1'},
+             'mochitest-browser-chrome-e10s-2': {'group': 'M-e10s', 'code': 'bc1'},
+             'mochitest-browser-chrome-e10s-3': {'group': 'M-e10s', 'code': 'bc1'},
              'mochitest-e10s-devtools-chrome': {'group': 'M-e10s', 'code': 'dt'},
              'mochitest-e10s-devtools-chrome-1': {'group': 'M-e10s', 'code': 'dt1'},
              'mochitest-e10s-devtools-chrome-2': {'group': 'M-e10s', 'code': 'dt2'},
              'mochitest-e10s-devtools-chrome-3': {'group': 'M-e10s', 'code': 'dt3'},
              'mochitest-e10s-devtools-chrome-4': {'group': 'M-e10s', 'code': 'dt4'},
+             'mochitest-devtools-chrome-e10s-1': {'group': 'M-e10s', 'code': 'dt4'},
+             'mochitest-devtools-chrome-e10s-2': {'group': 'M-e10s', 'code': 'dt4'},
+             'mochitest-devtools-chrome-e10s-3': {'group': 'M-e10s', 'code': 'dt4'},
+             'mochitest-devtools-chrome-e10s-4': {'group': 'M-e10s', 'code': 'dt4'},
              'mochitest-push': {'group': 'M', 'code': 'p'},
              'xpcshell': {'group': '', 'code': 'X'},
              'xpcshell-1': {'group': '', 'code': 'X1'},
```

Comment:
As I mentioned few times, I believe this data structure can be generated automatically.
It is pretty standard string manipulation.